### PR TITLE
added top_k argument in the run function of ElasticSearcBM25Retriever

### PR DESCRIPTION
**Pull request by sahusiddharth**
The top_k can only be defined at initialization, It would allow users to change the top_k at the pipeline runtime too.

### CODE REVIEW

File: integrations/elasticsearch/src/elasticsearch_haystack/bm25_retriever.py [hunk 1]
```diff
@@ -48,12 +48,12 @@ def from_dict(cls, data: Dict[str, Any]) -> "ElasticsearchBM25Retriever":
         return default_from_dict(cls, data)
 
     @component.output_types(documents=List[Document])
-    def run(self, query: str):
+    def run(self, query: str, top_k: int=None):
```
Comment (masci) [inline integrations/elasticsearch/src/elasticsearch_haystack/bm25_retriever.py:51]: `top_k` should be typed as `Optional`

File: integrations/elasticsearch/src/elasticsearch_haystack/bm25_retriever.py [hunk 2]
```diff
@@ -48,12 +48,12 @@ def from_dict(cls, data: Dict[str, Any]) -> "ElasticsearchBM25Retriever":
...
         docs = self._document_store._bm25_retrieval(
             query=query,
             filters=self._filters,
             fuzziness=self._fuzziness,
-            top_k=self._top_k,
+            top_k=self._top_k if top_k == None else top_k,
```
Comment (masci) [inline integrations/elasticsearch/src/elasticsearch_haystack/bm25_retriever.py:56]: you can simplify this statement with `top_k = top_k or self.top_k`

File: integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py [hunk 3]
```diff
@@ -64,17 +64,18 @@ def from_dict(cls, data: Dict[str, Any]) -> "ElasticsearchEmbeddingRetriever":
         return default_from_dict(cls, data)
 
     @component.output_types(documents=List[Document])
-    def run(self, query_embedding: List[float]):
+    def run(self, query_embedding: List[float], top_k:int = None):
```
Comment (masci) [inline integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py:67]: Same as above, `top_k` is optional

File: integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py [hunk 4]
```diff
@@ -64,17 +64,18 @@ def from_dict(cls, data: Dict[str, Any]) -> "ElasticsearchEmbeddingRetriever":
...
         """
         Retrieve documents using a vector similarity metric.
 
         :param query_embedding: Embedding of the query.
+        :param top_k: Maximum number of Documents to return
```
Comment (masci) [inline integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py:72]: Thanks for fixing the docs! It's out of the scope of this PR, but would you mind adding a similar docstring to the `run` method of the other retriever component?

**Review by masci — CHANGES_REQUESTED** [review]
Thanks for the PR! I left a couple of comments but it looks good. Would you mind singing the CLA before we merge this? Thanks in advance!

File: integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py (hunk 4 above)
Comment (sahusiddharth) [reply integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py:72]: @masci 

The changes you suggested will be done soon.

**Question:**  When you mentioned other retriever components you mean in the main haystack right?

Comment (masci) [reply integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py:72]: @sahusiddharth no I mean the bm25 retriever in this integration, see https://github.com/deepset-ai/haystack-core-integrations/blob/main/integrations/elasticsearch/src/elasticsearch_haystack/embedding_retriever.py#L69-L72

**Comment by sahusiddharth** [comment]
@masci All done

**Review by anakin87 — APPROVED** [review]
I made some little changes.
Thanks for this PR!

**Pull request merged**